f5/bigip/tm/gtm/datacenter.py

```diff
@@ -0,0 +1,96 @@
+# coding=utf-8
+#
+# Copyright 2014-2015 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""BIG-IP® Global Traffic Manager (GTM) datacenter module.
+
+REST URI
+    ``http://localhost/mgmt/tm/gtm/datacenter``
+
+GUI Path
+    ``DNS --> GSLB : Data Centers``
+
+REST Kind
+    ``tm:gtm:datacenter:*``
+"""
+
+from f5.bigip.mixins import ExclusiveAttributesMixin
+from f5.bigip.resource import Collection
+from f5.bigip.resource import Resource
+
+
+class Datacenters(Collection):
+    """BIG-IP® GTM datacenter collection"""
+    def __init__(self, gtm):
+        super(Datacenters, self).__init__(gtm)
+        self._meta_data['allowed_lazy_attributes'] = [Datacenter]
+        self._meta_data['attribute_registry'] =\
+            {'tm:gtm:datacenter:datacenterstate': Datacenter}
+
+
+class Datacenter(Resource, ExclusiveAttributesMixin):
+    """BIG-IP® GTM datacenter resource"""
+    def __init__(self, dc_s):
+        super(Datacenter, self).__init__(dc_s)
+        self._meta_data['required_json_kind'] =\
+            'tm:gtm:datacenter:datacenterstate'
+        self._meta_data['exclusive_attributes'].append(('enabled', 'disabled'))
+
+    def _endis_able(self, config_dict):
+        if 'enabled' in config_dict and not config_dict['enabled']:
+            config_dict['disabled'] = True
+        elif 'disabled' in config_dict and not config_dict['disabled']:
+            config_dict['enabled'] = True
+        return config_dict
+
+    def _endis_attrs(self):
+        if 'disabled' in self.__dict__:
+            self.__dict__['enabled'] = not self.__dict__['disabled']
+        if 'enabled' in self.__dict__:
+            self.__dict__['disabled'] = not self.__dict__['enabled']
+
+    def create(self, **kwargs):
+        kwargs = self._endis_able(kwargs)
+
+        if 'enabled' in kwargs and kwargs['enabled']:
```

Comment:
I'm reasonably certain that this is handled by the `ExclusiveAttributesMixin`...